### PR TITLE
fuzz: libprotobuf-mutator integration.

### PR DESCRIPTION
**Pull request by htuch**
This PR adds libprotobuf-mutator as an external dep and integrates it
with the fuzz infrastructure. Some slight reworking of how fuzz tests
was done to make it consistent between regular fuzz and proto fuzz
tests.

Also fixed a small bug in ValueUtil::equal that was exposed via the fuzz
test.

Risk Level: Low
Testing: New example trivial fuzz test for ValueUtil::equal.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: bazel/external/libprotobuf_mutator.BUILD [hunk 1]
```diff
@@ -0,0 +1,27 @@
+cc_library(
```
Comment (htuch) [inline bazel/external/libprotobuf_mutator.BUILD]: @jmillikin-stripe I'm not super happy with how this turned out, I don't like non-trivial BUILD files in Envoy, but this was simplest vs. getting protobuf as a Bazel dep and the cmake build system to play nice. I've filed https://github.com/google/libprotobuf-mutator/issues/91 to request that we get proper upstream Bazel support for this project. I think the current solution is fine as this is a test-only dependency and it's a pretty simple library, we don't care about tests in this case.

Comment (jmillikin-stripe) [reply bazel/external/libprotobuf_mutator.BUILD]: You might be able to simplify this with globs:
```
srcs = glob(["**/*.c", "**/*.h"], exclude=["**/*_test.cc"])
```

Comment (htuch) [reply bazel/external/libprotobuf_mutator.BUILD]: @jmillikin-stripe can you take another look at what I've done for OS X? I'm not very happy with it, I would like something more like https://github.com/bazelbuild/bazel/issues/3510.

**Comment by htuch** [comment]
Reassigning to @mattklein123 since he wants to learn a bit about proto fuzzing and Dan is OOTO.

**Pull request merged**